src/connection/Connection.ts

```diff
@@ -517,10 +517,26 @@ export class Connection {
         const migrations = connectionMetadataBuilder.buildMigrations(this.options.migrations || []);
         ObjectUtils.assign(this, { migrations: migrations });
 
-        this.driver.database = <string> this.options.database;
+        this.driver.database = this.getDatabaseName();
 
         // validate all created entity metadatas to make sure user created entities are valid and correct
         entityMetadataValidator.validateMany(this.entityMetadatas.filter(metadata => metadata.tableType !== "view"), this.driver);
     }
 
+    // This database name property is nested for replication configs.
+    protected getDatabaseName(): string {
+    const options = this.options;
```

Comment:
```typescript
const {replication, database} = this.options;
// ...
```

Reply:
This causes an issue with the typeguard. I can't be sure `this.options` contains `replication` until after the guard. I was trying to avoid the use of `any`. I like destructuring, but don't think it's worth the cost here.